hw3/React-App/src/Pages/Search.js

```diff
@@ -0,0 +1,61 @@
+import { useState, useEffect } from "react";
+import "./Search.css";
+export default function Search() {
+  const url = "https://thronesapi.com/api/v2/Characters";
+  const [characters, setCharacters] = useState([]);
+  const [filteredCharacters, setFilteredCharacters] = useState([]);
+  const [searched, setSearched] = useState(false); // new state variable
+  useEffect(() => {
+    fetch(url)
+      .then((response) => response.json())
+      .then((data) => {

```

Comment:
Calls the same API and fetches the same data as that of Charts.js

Fetching the data (calling the exact same API) twice, in both the Charts and Search components is inefficient. The data they both rely on is the same, and additionally both of these components have the same parent.

So why not let the data flow from parent to child?

It would make a lot more sense to call the API once in the App component, then pass the data as props to the Charts and Search components. This would be more efficient since the API would only need to be called once, but the data would be available across the components that need it.

In summary, you can just fetch the data once and then pass it to child components via props.